src/corelibs/U2View/src/phyltree/CreatePhyTreeDialogController.cpp

```diff
@@ -171,7 +172,11 @@ bool CreatePhyTreeDialogController::checkMemory() {
 
 void CreatePhyTreeDialogController::initSaveController(const MultipleSequenceAlignmentObject *msaObject) {
     SaveDocumentControllerConfig config;
-    config.defaultFileName = GUrlUtils::getNewLocalUrlByExtention(msaObject->getDocument()->getURLString(), msaObject->getGObjectName(), ".nwk", "");
+    QString urlStr = msaObject->getDocument()->getURLString();
+    if (!FileAndDirectoryUtils::isDirectoryWritable(QFileInfo(urlStr).dir().path())) {
+        urlStr = GUrlUtils::getDefaultDataPath() + "/" + msaObject->getGObjectName();
```

Comment:
I suggest write a message to log in this case, for example:

`coreLog.details(tr("You have no write access to the \"%1\" directory, the result will be written to \"%2\".").arg(urlStr).arg(path));`

Reply:
Also I think it's better to save result to "data/samples/Newick", not just "data"